webui/react/src/components/CheckpointDetails.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { PropsWithChildren, useCallback } from 'react';
+
+import useModalCheckpoint from 'hooks/useModal/useModalCheckpoint';
+import {
+  CheckpointDetail, CheckpointWorkloadExtended, ExperimentBase,
+} from 'types';
+
+interface Props {
+  checkpoint: CheckpointWorkloadExtended | CheckpointDetail;
+  experiment: ExperimentBase;
+  title: string;
+}
+
+const CheckpointDetails: React.FC<Props> = (
+  {
+    checkpoint,
+    experiment,
+    title,
+    ...props
+  }: PropsWithChildren<Props>,
+) => {
+  const { modalOpen: openModalDelete } =
+  useModalCheckpoint({ checkpoint: checkpoint, config: experiment.config, title: title });
+  const handleModalCheckpointClick = useCallback(() => openModalDelete(), [ openModalDelete ]);
+
+  return (
+    <span onClick={() => handleModalCheckpointClick()}>
+      {props.children}
```

Comment:
**suggestion:** what do you think about having the default content defined as...
```
<Tooltip title="View Checkpoint">
  <Button
    aria-label="View Checkpoint"
    icon={<Icon name="checkpoint" />}
  />
</Tooltip>
```
if `props.children` is undefined?

Reply:
I have made this update.